antarest/study/storage/variantstudy/model/command_listener/command_listener.py

```diff
@@ -0,0 +1,33 @@
+# Copyright (c) 2024, RTE (https://www.rte-france.com)
+#
+# See AUTHORS.txt
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# SPDX-License-Identifier: MPL-2.0
+#
+# This file is part of the Antares project.
+
+from abc import ABC, abstractmethod
+
+from antarest.core.serialization import AntaresBaseModel
+
+
+class ICommandListener(ABC, AntaresBaseModel, extra="forbid", arbitrary_types_allowed=True):
+    """
+    Interface for all listeners that can be given inside the `apply` method of a command.
+    """
+
+    task_id: str
```

Comment:
If you define a setter for this variable, you can add a underscore like _task_id: str
And also add a property : 
    @property
    def task_id(self) -> str:
        return self._task_id

Reply:
Indeed I'll do that

Reply:
In fact Pydantic doesn't handle @property really well so I can't validate the model against task_id :/ If you find a way to do so please feel free to provide the way to do it, else I'll keep the old implementation